### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by v4hn** [AIS-Bonn/TopiCo#8]
Hey Marius,

I've basically just found this repository and did not use it (yet?). But I'll have to ask out of pure curiosity:

The repository includes a [`LICENCE.md`](https://github.com/AIS-Bonn/TopiCo/blob/master/LICENSE.md) for BSD-3-Clause, but the headers refer to a ["Student License"](https://github.com/AIS-Bonn/TopiCo/blob/master/ROS/topico/include/topico_wrapper.h#L1-L9), although some of them also include a BSD header somewhere in the file.

What license do you want to publish this code under?

**Comment by MariusBeul** [comment]
Hey v4hn,

thank you for asking! The MATLAB source is licensed with the BSD-3-Clause license.

But since I generated the CPP version with the Student Version of MATLAB Coder, MATLAB Coder inserts these additional headers into the CPP sources. To get rid of these headers, I suggest to use a non-student version of MATLAB Coder to regenerate the CPP sources. See [this issue](https://github.com/AIS-Bonn/TopiCo/issues/6) for a tutorial.

Best
Marius

**Comment by v4hn** [comment]
Thank you for explaining! I'm not familiar with MATLAB Coder and wrongly assumed the C++ code was an independent component.

It would definitely improve the usefulness of this repository in ROS if someone were to contribute generated code with a full license.
Maybe you should leave an issue open until someone did that because the BSD-3-Clause indicator for this repository is technically wrong at the moment. :/

Though personally I'm happy to use "academia-only" code as well. :-)

**Comment by MariusBeul** [comment]
Thanks for the suggestion. I was able to update from the student version to the academic version which in addition to academic research and meeting course requirements at degree granting institutions also allows use in teaching.
I will reopen the issue as suggested.

TLDR: If you have a commercial version of MATLAB, please follow [this procedure](https://github.com/AIS-Bonn/TopiCo/issues/6) to generate CPP files from the MATLAB sources and provide them to the public!
